src/js/utils/ResourcesUtil.js

```diff
@@ -0,0 +1,127 @@
+import CompositeState from '../structs/CompositeState';
+import Node from '../structs/Node';
+import StringUtil from './StringUtil';
+
+const DefaultResourceTypes = {
+  cpus: {label: 'CPU', colorID: 0},
+  mem: {label: 'Memory', colorID: 6},
+  disk: {label: 'Disk', colorID: 3}
+};
+
+const usedColors = Object.keys(DefaultResourceTypes)
+.map(function (resource) {
+  return DefaultResourceTypes[resource].colorID;
+});
+const availableColors = Array(9).fill().map(function (value, index) {
+  return index;
+}).filter(function (value) {
+  return usedColors.indexOf(value) === -1;
+});
+
+function getAvailableColors() {
+  let colors = availableColors.slice(0);
+  // Concat the array twice. This is in case there's many resource.
+  // Unlikely that there will more than 4 anyways.
+  return colors.concat(colors);
+}
+
+const ResourcesUtil = {
+  getDefaultResources: function () {
+    return Object.keys(DefaultResourceTypes);
+  },
+
+  getAvailableResources: function (excludeList = []) {
+    let item = CompositeState.getNodesList().getItems()[0];
+
+    if (!item) {
+      item = CompositeState.getServiceList().getItems()[0];
+
+      if (!item) {
+        let fakeUsedResources = this.getDefaultResources()
+        .reduce(function (memo, resource) {
+          memo[resource] = 0;
+          return memo;
+        }, {});
+        item = new Node({used_resources: fakeUsedResources});
+      }
+    }
+
+    let resources = Object.keys(item.get('used_resources'));
+
+    if (excludeList.length) {
+      resources = resources.filter(function (resource) {
+        // If it's not in the exclude list, we want it
+        return excludeList.indexOf(resource) === -1;
+      });
+    }
+
+    return resources;
+  },
+
+  getAdditionalResources: function () {
+    let rest = this.getAvailableResources(this.getDefaultResources());
+    // We sort so we get the same color in future calls to this method
+    rest.sort();
+    return rest;
+  },
+
+  getResourceLabel: function (resource) {
+    if (DefaultResourceTypes[resource]) {
+      return DefaultResourceTypes[resource].label;
+    }
+
+    if (resource.length <= 3) {
+      return resource.toUpperCase();
+    } else {
+      return StringUtil.capitalize(resource.toLowerCase());
+    }
+  },
+
+  getResourceLabels: function () {
+    let resources = this.getAvailableResources();
+
+    return resources.reduce((memo, resource) => {
+      memo[resource] = this.getResourceLabel(resource);
+      return memo;
```

Comment:
ah so you're jumping onto the `reduce` wagon. that's cool. can you put a newline before `return`?